src/Symfony/Component/HttpFoundation/Session/Storage/Proxy/SessionHandlerProxy.php

```diff
@@ -50,16 +50,16 @@ public function open($savePath, $sessionName)
     #[\ReturnTypeWillChange]
     public function close()
     {
-        return (bool) $this->handler->close();
+        return $this->handler->close();
     }
 
     /**
-     * @return string
+     * @return string|false
```

Comment:
https://www.php.net/manual/en/sessionhandlerinterface.read.php

Reply:
But, right now, we guarantee a string, so this is a BC break.

Reply:
We did exactly the same for `gc` (https://github.com/symfony/symfony/pull/41495#discussion_r643126681). And from what I understood, PHP itself uses the returned value?

Reply:
I think we should do this change because this is a proxy, and the proxy is not supposed to alter the return types of the wrapped handler. I really really doubt the change will affect anybody since 1. this is a proxy and ppl shouldn't code against the behavior of the proxy but should code against the interface and 2. the interface is designed to integrate with the PHP engine.